### PR TITLE
Add BMH annotation keys to docs

### PR DESCRIPTION
**Pull request by carbonin**
Previously we said which fields could be set using annotations, but didn't document the annotation keys

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [x] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

### CODE REVIEW

File: docs/hive-integration/README.md [hunk 1]
```diff
@@ -160,10 +160,17 @@ In case that the Bare Metal Operator is installed, the Baremetal Agent Controlle
 
 - Find the right pairs of BMH/Agent using their MAC addresses
 - Set the Image.URL in the BMH copying it from the InfraEnv's status.
```
Comment (omertuc) [inline docs/hive-integration/README.md:162]: Is this even still true since converged?

Comment (omertuc) [reply docs/hive-integration/README.md:162]: (unrelated to PR)

Comment (carbonin) [reply docs/hive-integration/README.md:162]: No, but there's a specific section for converged.

**Pull request merged**